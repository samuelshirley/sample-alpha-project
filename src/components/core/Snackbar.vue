<template>
  <v-snackbar
    v-model="model"
    :color="snackbar.type"
    :timeout="snackbar.timeout"
    right
  >
    <v-icon
      :dark="snackbar.type !== 'warning'"
      v-text="icon"
    />
    <span
      :class="snackbar.type === 'warning' ? 'black--text' : 'white--text'"
      v-text="snackbar.msg"
    />
    <v-btn
      v-if="snackbar.close"
      class="ml-1"
      icon
      flat
      @click="model = false"
    >
      <v-icon
        :dark="snackbar.type !== 'warning'"
        small
        v-text="'mdi-close'"
      />
    </v-btn>
  </v-snackbar>
</template>

<script>
// Utilities
import {
  mapState,
} from 'vuex'

const ICON_MAP = {
  error: 'mdi-alert-octagon',
  info: 'mdi-information',
  success: 'mdi-check-circle',
  warning: 'mdi-alert-circle',
}

export default {
  data: () => ({
    model: false,
  }),

  computed: {
    ...mapState('app', ['snackbar']),
    icon () {
      return ICON_MAP[this.snackbar.type] || 'mdi-playlist-check'
    },
  },

  watch: {
    snackbar () {
      this.model = true
    },
  },
}
</script>
