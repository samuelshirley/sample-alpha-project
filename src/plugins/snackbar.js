import Vue from 'vue'

Vue.mixin({
  computed: {
    $snackbar () {
      return {
        default: (m, t, c) => this.$message('', m, t, c),
        error: (m, t, c) => this.$message('error', m, t, c),
        info: (m, t, c) => this.$message('info', m, t, c),
        success: (m, t, c) => this.$message('success', m, t, c),
        warning: (m, t, c) => this.$message('warning', m, t, c),
      }
    },
  },

  methods: {
    $message (type, msg, timeout = 6000, close = false) {
      this.$store.commit('app/setSnackbar', {
        type,
        msg,
        timeout,
        close,
      })
    },
  },
})
