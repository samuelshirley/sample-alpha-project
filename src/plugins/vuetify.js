import Vue from 'vue'
import Vuetify from 'vuetify/lib'
import 'vuetify/src/stylus/app.styl'

Vue.use(Vuetify, {
  theme: {
    primary: '#1867c0',
    secondary: '#5CBBF6',
    tertiary: '#E57373',
    accent: '#005CAF',
    error: '#FF5252',
    info: '#2196F3',
    success: '#4CAF50',
    warning: '#FFC107',
  },
  iconfont: 'mdi',
})
