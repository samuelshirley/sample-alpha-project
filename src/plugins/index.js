import '@/plugins/axios'
import '@/plugins/snackbar'
import '@/plugins/vuetify'
