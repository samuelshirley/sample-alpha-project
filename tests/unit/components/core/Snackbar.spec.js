import Vue from 'vue'
import Vuex from 'vuex'
import { mount } from '@vue/test-utils'
import appState from '@/store/modules/app/state'
import appMutations from '@/store/modules/app/mutations'
import Snackbar from '@/components/core/Snackbar'
import '@/plugins/snackbar'

Vue.use(Vuex)

describe('Snackbar.vue', () => {
  let store
  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        app: {
          namespaced: true,
          state: appState,
          mutations: appMutations,
        },
      },
    })
  })

  it('should match snapshot', async () => {
    const wrapper = mount(Snackbar, {
      store,
    })
    // wrapper.vm.model = true
    wrapper.vm.$store.commit('app/setSnackbar', {
      type: 'error',
      timeout: 0,
      msg: 'Error Message',
      close: false,
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
