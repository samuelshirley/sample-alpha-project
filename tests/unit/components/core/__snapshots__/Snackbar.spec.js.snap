// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar.vue should match snapshot 1`] = `
<div class="v-snack v-snack--active v-snack--bottom v-snack--right" name="v-snack-transition">
  <div class="v-snack__wrapper error">
    <div class="v-snack__content"><i aria-hidden="true" class="v-icon mdi mdi-alert-octagon theme--dark"></i> <span class="white--text">Error Message</span>
      <!---->
    </div>
  </div>
</div>
`;
